UnitTestBot/UTBotJava
# UTBot multi-IDE support

## Description

### IDEs
Supported plugin building for PyCharm (Community/Professional) and GoLand.
![image](https://github.com/UnitTestBot/UTBotJava/assets/23080942/a277809c-9275-417a-9c53-ba20c0204d41)

By default plugin will be support languages according to `ideType` (parameter in `gradle.properties`):

| ideType | UTBot language support | IDE |
|-----------|------------|---|
| IC | Java, Python | IDEA Community |
| IU | Java, Python, JavaScript, Go | IDEA Ultimate |
| PC | Python | PyCharm Community |
| PY | Python, JavaScript | PyCharm Professional |
| GO | Go | GoLand |

For example, if you set `ideType=PC` (in `gradle.properties`) to build utbot plugin for PyCharm Community.

<details>
  <summary>Language to IDEs</summary>
 
| Language | ideTypes |
|-----------|------------|
| Java | IC, IU |
| Python | IC, IU, PC, PY |
| JavaScript | IU, PY |
| Go | IU, GO |
  
</details>

__Additional functionality:__ you can set `projectType=Community` to build plugin without Go and JavaScript for IDEA Ultimate.

__NB:__ Plugin for IU works in IC (only for Java and Python), for IC works in IU; plugin for PY works in PC and for PC works in PY.

### Settings

UTBot settings have been separated to "common" and <language>-specific (now only java-specific). Java-specific settings have been moved to UnitTestBot sub-page.

In PyCharm or GoLand build Java settings will not be included.

Also `Experimental language support` settings now is enabled by default.

## How to test

### Manual tests

Build plugin (`$ gradle buildPlugin`) with different `ideType` and check that UTBot supports all suitable languages.

## Self-check list
